src/draw/drawLabels.ts

```diff
@@ -0,0 +1,172 @@
+import { Canvas } from 'skia-canvas';
+
+import type { Image, ImageDataArray } from '../Image.js';
+import type { Point } from '../geometry/index.js';
+import { validateValues } from '../utils/validators/validators.js';
+
+export type Label = number | string;
+
+export interface DrawLabelsOptions {
+  /**
+   *  Size and type of font.
+   */
+  font?: string;
+  /**
+   *  Font color.Should be in rgba8 format.
+   */
+  fontColor?: number[];
+}
+/**
+ * Draws different labels on images.
+ * @param image - Image to draw labels on.
+ * @param labels - Labels to draw.
+ * @param coordinates - Coordinates where to draw labels.
+ * @param options - DrawLabelsWithCanvasOptions.
+ * @returns new image with drawn labels on it.
+ */
+export function drawLabels(
+  image: Image,
+  labels: Label[],
+  coordinates: Point[],
+  options: DrawLabelsOptions = {},
+) {
+  const newImage = image.clone();
+  const canvas = new Canvas(image.width, image.height);
+  const { font = '12px Helvetica', fontColor = [255, 255, 255, 255] } = options;
+
+  validateValues(fontColor, image);
+  if (coordinates.length === 0) {
+    throw new RangeError('You must specify at least one label coordinate.');
+  }
+  if (labels.length !== coordinates.length) {
+    throw new Error('Positions and labels must be arrays of the same size.');
+  }
+  const alpha = fontColor[3] ? fontColor[3] / 255 : 1;
+  const normalizedColor = [
+    fontColor[0] ?? 255,
+    fontColor[1] ?? 255,
+    fontColor[2] ?? 255,
+    alpha,
+  ];
+
+  const ctx = canvas.getContext('2d');
+  ctx.font = font;
+  ctx.fillStyle = `rgba(${normalizedColor.join(',')})`;
+  for (let i = 0; i < labels.length; i++) {
+    const coordinate = coordinates[i % coordinates.length];
```

Comment:
I don't understand what the point is to have label and coordinates in different arrays and reuse coordinates across labels.